### DISCUSSION TITLE
测试 #37 支持QQ频道版本(v0.1.13以上可用）

### DISCUSSION THREAD

**Discussion by leecoooo** [eric2788/vup_monitors#41] [Q&A] [answered]
测试环境：
go-cqhttp v1.0.0-rc4
biligo-live-ws v0.1.15
vup_monitors v0.1.13alpha

直播间监控新增没问题
添加高亮没问题
测试高亮弹幕提醒：
3|vup      | info: go-cqhttp 连接成功。 {"timestamp":"2023-03-03T17:03:08.836Z"}
3|vup      | info: 离线数据储存已更新。 {"timestamp":"2023-03-03T17:03:19.097Z"}
3|vup      | info: 房间进入通知已发送给 1 个QQ群组。 {"timestamp":"2023-03-03T17:04:56.051Z"}
3|vup      | info: 高亮弹幕通知已发送给 1 个QQ群组。 {"timestamp":"2023-03-03T17:05:10.462Z"}
3|vup      | info: 高亮弹幕通知已发送给 1 个QQ群组。 {"timestamp":"2023-03-03T17:07:09.337Z"}
3|vup      | info: 离线数据储存已更新。 {"timestamp":"2023-03-03T17:08:10.154Z"}
3|vup      | info: 离线数据储存已更新。 {"timestamp":"2023-03-03T17:08:22.292Z"}
3|vup      | info: 高亮弹幕通知已发送给 1 个QQ群组。 {"timestamp":"2023-03-03T17:08:30.370Z"}
3|vup      | info: 离线数据储存已更新。 {"timestamp":"2023-03-03T17:11:30.752Z"}
3|vup      | info: 离线数据储存已更新。 {"timestamp":"2023-03-03T17:11:51.834Z"}
3|vup      | info: 高亮弹幕通知已发送给 1 个QQ群组。 {"timestamp":"2023-03-03T17:12:00.194Z"}

gc log无反应
实际未发送QQ频道消息

频道号码-子频道号码，是这个 “1087781643123732-215811651” 格式吗？

**Comment by eric2788** [comment]
交給你測試了

**Comment by eric2788** [comment, marked as answer]
新推出了 alpha 版本，你可以試試

**Reply by eric2788** [reply]
https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha

**Reply by leecoooo** [reply]
> https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha

用了这个版本也是发送频道消息没反应

**Reply by eric2788** [reply]
@Marxel-cn

**Reply by Marxel-cn** [reply]
> > https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha
> 
> 用了这个版本也是发送频道消息没反应
我之前自测是ok的，问下你是怎么获取频道和子频道号的？

**Reply by leecoooo** [reply]
> > > https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha
> > 
> > 
> > 用了这个版本也是发送频道消息没反应
> > 我之前自测是ok的，问下你是怎么获取频道和子频道号的？

在gc的log里面获取的，gc会把频道的消息都显示在log里面，里面就有频道和子频道的号码。

**Reply by Marxel-cn** [reply]
> > > > https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha
> > > 
> > > 
> > > 用了这个版本也是发送频道消息没反应
> > > 我之前自测是ok的，问下你是怎么获取频道和子频道号的？
> 
> 在gc的log里面获取的，gc会把频道的消息都显示在log里面，里面就有频道和子频道的号码。

先重开一次gc试试看？

**Reply by leecoooo** [reply]
> > > > > https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha
> > > > 
> > > > 
> > > > 用了这个版本也是发送频道消息没反应
> > > > 我之前自测是ok的，问下你是怎么获取频道和子频道号的？
> > 
> > 
> > 在gc的log里面获取的，gc会把频道的消息都显示在log里面，里面就有频道和子频道的号码。
> 
> 先重开一次gc试试看？

log内容是：[2023-03-09 10:26:02] [INFO]: 收到来自频道 xxx(71087781643123732) 子频道 xxx(2711367) 内 xxx(144115218769825846) 的消息: xxxx

**Reply by leecoooo** [reply]
重启log
1|gc  | [2023-03-09 10:49:45] [INFO]: 开始尝试登录并同步消息...
1|gc  | [2023-03-09 10:49:45] [INFO]: 使用协议: iPad
1|gc  |
1|gc  | [2023-03-09 10:49:45] [INFO]: Protocol -> connect to server: 183.47.117.157:8080
1|gc  |
1|gc  | [2023-03-09 10:49:45] [WARNING]: Protocol -> device lock is disable. http api may fail.
1|gc  |
1|gc  | [2023-03-09 10:49:48] [INFO]: 登录成功 欢迎使用: xxx
1|gc  | [2023-03-09 10:49:48] [INFO]: 开始加载好友列表...
1|gc  |
1|gc  | [2023-03-09 10:49:48] [INFO]: 共加载 4 个好友.
1|gc  | [2023-03-09 10:49:48] [INFO]: 开始加载群列表...
1|gc  |
1|gc  | [2023-03-09 10:49:48] [INFO]: 收到服务器地址更新通知, 将在下一次重连时应用.
1|gc  |
1|gc  | [2023-03-09 10:49:50] [INFO]: 共加载 6 个群.
1|gc  |
1|gc  | [2023-03-09 10:49:50] [INFO]: 资源初始化完成, 开始处理信息.
1|gc  | [2023-03-09 10:49:50] [INFO]: アトリは、高性能ですから!
1|gc  |
1|gc  | [2023-03-09 10:49:50] [INFO]: 正在检查更新.
1|gc  |
1|gc  | [2023-03-09 10:49:50] [INFO]: CQ HTTP 服务器已启动: 127.0.0.1:5700
1|gc  |
1|gc  | [2023-03-09 10:49:50] [INFO]: CQ WebSocket 服务器已启动: 127.0.0.1:6700
1|gc  |
1|gc  | [2023-03-09 10:49:51] [INFO]: 检查更新完成. 当前已运行最新版本.
1|gc  | [2023-03-09 10:49:51] [INFO]: 开始诊断网络情况
1|gc  |
1|gc  | [2023-03-09 10:49:53] [INFO]: 网络诊断完成. 未发现问题

**Reply by leecoooo** [reply]
> > > > > https://github.com/eric2788/vup_monitors/releases/tag/0.1.13.alpha
> > > > 
> > > > 
> > > > 用了这个版本也是发送频道消息没反应
> > > > 我之前自测是ok的，问下你是怎么获取频道和子频道号的？
> > 
> > 
> > 在gc的log里面获取的，gc会把频道的消息都显示在log里面，里面就有频道和子频道的号码。
> 
> 先重开一次gc试试看？

没啥效果，还是一样

**Reply by Marxel-cn** [reply]
> log内容是：[2023-03-09 10:26:02] [INFO]: 收到来自频道 xxx(71087781643123732) 子频道 xxx(2711367) 内 xxx(144115218769825846) 的消息: xxxx


我用0.1.13a的包在linux下测试正常
[2023-03-09 11:45:46] [INFO]: 发送频道 xxx(625216973979765461) 子频道 xxx(144442824) 的消息: xxx ... (10-2083536598)

配置是这样的：
{"blive":{"highlight":{"123":[],"625216973979765461-144442824":[xxx,xxx]},"highlight_private":{"123":[]},"focus_users":{"123":[]},"subscribing":[xxx,xxx]}}

对你上面的情况，应该就是在highlight里配置
"71087781643123732-2711367":[高亮用户uid]

win10下的测试
[2023-03-09 11:53:30] [INFO]: 发送频道 xxx(625216973979765461) 子频道 xxx(144442824) 的消息: xxx ... (12-2066269723)

**Reply by leecoooo** [reply]
收到，我看看配置文件。不过我是用指令加的会不会有问题？

**Reply by Marxel-cn** [reply]
> 收到，我看看配置文件。不过我是用指令加的会不会有问题？

先改配置文件吧，指令添加的时候因为群号会被转成纯数字，我试了下会导致-和后面的数字丢失，之后我会提个PR改掉这部分校验

**Reply by leecoooo** [reply]
> > 收到，我看看配置文件。不过我是用指令加的会不会有问题？
> 
> 先改配置文件吧，指令添加的时候因为群号会被转成纯数字，我试了下会导致-和后面的数字丢失，之后我会提个PR改掉这部分校验

确实是这样，成功了，感谢

**Reply by Marxel-cn** [reply]
> > > 收到，我看看配置文件。不过我是用指令加的会不会有问题？
> > 
> > 
> > 先改配置文件吧，指令添加的时候因为群号会被转成纯数字，我试了下会导致-和后面的数字丢失，之后我会提个PR改掉这部分校验
> 
> 确实是这样，成功了，感谢

#42

**Comment by Marxel-cn** [comment]
多嘴一句，我最早是想在配置里新增一个频道项目，然后把频道和群区分开的，后来发现频道那边的用户号跟qq号是完全分离的，所以想要响应频道控制消息的话还要继续加配置项，就决定只做发送了，然后频道又有频道号，所以就简单点直接用-来判断是群号还是频道-子频道号了

**Comment by eric2788** [comment]
@Marxel-cn 你有沒有tg qq之類的通訊工具，加一下我，想加你做cooperator
